reducing_utils.go

```diff
@@ -74,23 +82,26 @@ References:
 	http://www.oasis-open.org/committees/documents.php?wg_abbrev=office-formula
 	OpenDocument-formula-20090508.odt
 */
-func IPmt(rate float64, per int64, nper int64, pv float64, fv float64, when paymentperiod.Type) float64 {
+func IPmt(rate decimal.Decimal, per int64, nper int64, pv decimal.Decimal, fv decimal.Decimal, when paymentperiod.Type) *decimal.Decimal {
+	// TODO: update nan and remove any rounding here.
 	totalPmt := Pmt(rate, nper, pv, fv, when)
-	ipmt := rbl(rate, per, totalPmt, pv, when) * rate
+	one := decimal.NewFromInt(1)
+	ipmt := rbl(rate, per, totalPmt, pv, when).Mul(rate)
 	if when == paymentperiod.BEGINNING {
 		if per < 1 {
-			return math.NaN()
```

Comment:
create decimal from math.NaN() and return?

Reply:
it panics on nan init for decimal. For decimal case, it should never be nan, if somehow it is, it will panic( decimal pkg handles nan error by panic() ).

Reply:
@krantideep95 we can handle it similar to how `Nper()` is handled.